helpers/session.go

```diff
@@ -0,0 +1,82 @@
+package helpers
+
+import (
+	"go-google-scraper-challenge/helpers/log"
+	"go-google-scraper-challenge/lib/models"
+
+	"github.com/gin-contrib/sessions"
+	"github.com/gin-gonic/gin"
+)
+
+const CurrentUserKey = "CURRENT_USER_ID"
+const (
+	FlashTypeSuccess = "success"
+	FlashTypeInfo    = "info"
+	FlashTypeError   = "error"
+)
+
+type Flash struct {
+	Type    string
+	Message string
+}
+
+func GetCurrentUser(ctx *gin.Context) *models.User {
+	session := sessions.Default(ctx)
+	currentUserID := session.Get(CurrentUserKey)
+	if currentUserID == nil {
+		return nil
+	}
+
+	user, err := models.GetUserByID(currentUserID.(int64))
+	if err != nil {
+		log.Error("Fail to get current user", err.Error())
+
+		return nil
+	}
+
+	return user
+}
+
+func SetCurrentUser(ctx *gin.Context, userID int64) {
+	session := sessions.Default(ctx)
+	session.Set(CurrentUserKey, userID)
+	err := session.Save()
+	if err != nil {
+		log.Error("Fail to set current user", err.Error())
+	}
+}
+
+func SetFlash(ctx *gin.Context, flashType string, flashMessage string) {
+	session := sessions.Default(ctx)
+	session.AddFlash(flashMessage, flashType)
```

Comment:
@carryall Can you explain more about using flash here?
From my understanding about flash is a thing containing message to the frontend 😉

Reply:
Yes exactly, it's the notification message that will be shown on the UI after the user submit the form, on the technical part it'll be kept in the cookie and I added 3 types of them to display in different colors